service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/UploadCMOService.java

```diff
@@ -163,15 +163,23 @@ public UploadCMOEvent buildEventToPublish(CaseData caseData, CaseData caseDataBe
     }
 
     private void migrateDocuments(List<Element<HearingFurtherEvidenceBundle>> evidenceBundles, UUID selectedHearingId,
-                                  HearingBooking hearing, List<Element<SupportingEvidenceBundle>> supportingDocs) {
+                                  HearingBooking hearing, List<Element<SupportingEvidenceBundle>> cmpSupportingDocs) {
         Optional<Element<HearingFurtherEvidenceBundle>> bundle = findElement(selectedHearingId, evidenceBundles);
 
         if (bundle.isPresent()) {
-            bundle.get().getValue().getSupportingEvidenceBundle().addAll(supportingDocs);
+            List<Element<SupportingEvidenceBundle>> hearingDocs = bundle.get().getValue().getSupportingEvidenceBundle();
+
+            cmpSupportingDocs.forEach(element -> {
+                hearingDocs.stream()
```

Comment:
would it be better to only add those that aren't present? Or are you trying to keep them all grouped in the collection?

something like
```java
cmpSupportingDocs.stream()
                .filter(doc -> hearingDocs.stream().anyMatch(hearingDoc -> hearingDoc.getId().equals(doc.getId())))
                .forEach(hearingDocs::add);
```

Reply:
If you add only new, then changes to existing supporting document wont be propagated to hearing bundle.

Reply:
Good point, if the replace one or update a field it probably won't change the element id